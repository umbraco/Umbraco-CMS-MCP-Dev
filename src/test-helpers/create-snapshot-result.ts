import { BLANK_UUID } from "@/constants/constants.js";
import { CallToolResult } from "@modelcontextprotocol/sdk/types.js";

export function createSnapshotResult(result: any, idToReplace?: string) {
  if (!result?.content) {
    return result;
  }

  function normalizeItem(i: any) {
    const item = { ...i, id: BLANK_UUID };
    if (item.parent) {
      item.parent = { ...item.parent, id: BLANK_UUID };
    }
    if (item.createDate) {
      item.createDate = "NORMALIZED_DATE";
    }
    if (item.publishDate) {
      item.publishDate = "NORMALIZED_DATE";
    }
    if (item.updateDate) {
      item.updateDate = "NORMALIZED_DATE";
    }
    return item;
  }

  return {
    ...result,
    content: result.content.map((item: any) => {
      if (item.type === "text") {
        if (idToReplace) {
          // For single item responses
          let text = item.text.replace(idToReplace, BLANK_UUID);
          try {
            const parsed = JSON.parse(text);
            if (parsed.createDate) {
              parsed.createDate = "NORMALIZED_DATE";
            }
            if (parsed.availableUntil) {
              parsed.availableUntil = "NORMALIZED_DATE";
            }
            if (parsed.publishDate) {
              parsed.publishDate = "NORMALIZED_DATE";
            }
            if (parsed.updateDate) {
              parsed.updateDate = "NORMALIZED_DATE";
            }
            if (parsed.variants && Array.isArray(parsed.variants)) {
              parsed.variants = parsed.variants.map((variant: any) => {
                if (variant.createDate) variant.createDate = "NORMALIZED_DATE";
                if (variant.publishDate)
                  variant.publishDate = "NORMALIZED_DATE";
                if (variant.updateDate) variant.updateDate = "NORMALIZED_DATE";
                return variant;
              });
            }
            text = JSON.stringify(parsed);
          } catch {}
          return {
            ...item,
            text,
          };
        } else {
          // For list responses
          const parsed = JSON.parse(item.text);
          if (Array.isArray(parsed)) {
            // Handle ancestors API response
            return {
              ...item,
              text: JSON.stringify(parsed.map(normalizeItem)),
            };
          }
          // Handle other list responses
          if (parsed.items) {
            parsed.items = parsed.items.map(normalizeItem);
          }
          if (parsed.variants && Array.isArray(parsed.variants)) {
            parsed.variants = parsed.variants.map((variant: any) => {
              if (variant.createDate) variant.createDate = "NORMALIZED_DATE";
              if (variant.publishDate) variant.publishDate = "NORMALIZED_DATE";
              if (variant.updateDate) variant.updateDate = "NORMALIZED_DATE";
              return variant;
            });
          }
          return {
            ...item,
            text: JSON.stringify(parsed),
          };
        }
      }
      return item;
    }),
  };
}

export function normalizeErrorResponse(result: CallToolResult): CallToolResult {
  if (
    Array.isArray(result.content) &&
    result.content[0]?.text &&
    typeof result.content[0].text === "string"
  ) {
    // Replace any traceId in the text with a normalized version
    result.content[0].text = result.content[0].text.replace(
      /00-[0-9a-f]{32}-[0-9a-f]{16}-00/g,
      "normalized-trace-id"
    );
  }
  return result;
}
