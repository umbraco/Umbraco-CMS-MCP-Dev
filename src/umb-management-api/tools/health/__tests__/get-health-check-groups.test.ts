import GetHealthCheckGroupsTool from "../get/get-health-check-groups.js";
import { createSnapshotResult } from "@/test-helpers/create-snapshot-result.js";
import { jest } from "@jest/globals";

const TEST_SKIP_VALUE = 0;
const TEST_TAKE_VALUE = 10;

describe("get-health-check-groups", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
  });

  it("should get health check groups", async () => {
    const result = await GetHealthCheckGroupsTool().handler({
      skip: TEST_SKIP_VALUE,
      take: TEST_TAKE_VALUE
    }, { signal: new AbortController().signal });

    expect(createSnapshotResult(result)).toMatchSnapshot();
  });

  it("should handle invalid parameters", async () => {
    const result = await GetHealthCheckGroupsTool().handler({
      skip: -1,
      take: -1
    }, { signal: new AbortController().signal });

    expect(result).toMatchSnapshot();
  });
});