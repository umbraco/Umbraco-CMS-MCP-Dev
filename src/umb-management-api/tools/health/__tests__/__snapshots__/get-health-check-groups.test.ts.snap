// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-health-check-groups should get health check groups 1`] = `
{
  "content": [
    {
      "text": "{"total":6,"items":[{"name":"Configuration","id":"00000000-0000-0000-0000-000000000000"},{"name":"Data Integrity","id":"00000000-0000-0000-0000-000000000000"},{"name":"Live Environment","id":"00000000-0000-0000-0000-000000000000"},{"name":"Permissions","id":"00000000-0000-0000-0000-000000000000"},{"name":"Security","id":"00000000-0000-0000-0000-000000000000"},{"name":"Services","id":"00000000-0000-0000-0000-000000000000"}]}",
      "type": "text",
    },
  ],
}
`;

exports[`get-health-check-groups should handle invalid parameters 1`] = `
{
  "content": [
    {
      "text": "{"total":6,"items":[]}",
      "type": "text",
    },
  ],
}
`;
